build-system/babel-plugins/babel-plugin-amd-custom-define/define-template.js

```diff
@@ -0,0 +1,52 @@
+// @ts-nocheck
+/* eslint-disable no-var */
+
+/* global __ARGS__ */
+/* global __MODULE_NAME__ */
+/* global __HAS_EXPORTS__ */
+/* global __ONLY_DEP__ */
+
+(function define(deps, callback) {
```

Comment:
This is an implementation of `define` that's included through a custom babel plugin.

We could instead use esbuild's `inject` option to import a general-purpose `define()` implementation. Since every binary bundles its own `define()`, using a custom babel plugin has the benefit that we can DCE it:

- most extensions only import `bento`:

  ```js
  function define(s, j) {
    self.BENTO = self.BENTO || {};
    var d = self.BENTO.bento || {c: []};
    d.m ? j(d.m) : d.c.push(j);
  }
  ```

- `bento.js` only exports:

  ```js
  function define(S, Q) {
    self.BENTO = self.BENTO || {};
    var R = Q;
    (Q = function ($) {
      R.apply(null, arguments);
      var Z = 'bento',
        J = (self.BENTO[Z] = self.BENTO[Z] || {});
      for (J.m = $; J.c && J.c.length > 0; ) J.c.pop()($);
    }
    Q({});
  })
  ```

- in case an extension imports multiple modules:

  ```js
  function define(f, E) {
    self.BENTO = self.BENTO || {};
    Promise.all(
      f.map(function (b) {
        var h = self.BENTO[b] || {c: []};
        return h.m || new Promise(function (z) {
          h.c.push(z);
        });
      })
    ).then(function (b) {
      E.apply(null, b);
    });
  }
  ```

- worst case is an extension that exports and imports:

  ```js
  function define(l, x) {
    self.BENTO = self.BENTO || {};
    var z = x;
    x = function (y) {
      z.apply(null, arguments);
      var b = 'v0/bento-fit-text-1.0',
        h = (self.BENTO[b] = self.BENTO[b] || {});
      for (h.m = y; h.c && h.c.length > 0; ) h.c.pop()(y);
    };
    Promise.all(
      l.map(function (y) {
        var b = self.BENTO[y] || {c: []};
        return b.m || new Promise(function (h) {
          b.c.push(h);
        });
      })
    ).then(function (y) {
      x.apply(null, y);
    });
  }
  ```